### PR TITLE
Add -Yno-imports / sbt 1.1.6

### PR DESCRIPTION
**Pull request by NeQuissimus**
- [X] Add `-Yno-imports`
- [x] Fix the world

### CODE REVIEW

File: project/ScalazBuild.scala [hunk 1]
```diff
@@ -32,7 +32,8 @@ object Scalaz {
     "-Ywarn-unused:_,-imports",
     "-Ywarn-value-discard",
     "-opt:l:inline",
-    "-opt-inline-from:<source>"
+    "-opt-inline-from:<source>",
+    "-Yno-imports"
```
Comment (hrhino) [inline project/ScalazBuild.scala]: if only there were a way this could have been a one-line diff!

Comment (NeQuissimus) [reply project/ScalazBuild.scala]: It was but then you made me scalafmt :D

**Comment by NeQuissimus** [comment]
OH LAWD

**Comment by NeQuissimus** [comment]
Hows was the build in #1823 ever green? Something must have happened?! I'll take a look tomorrow....

File: project/ScalazBuild.scala [hunk 2]
```diff
@@ -45,7 +46,8 @@ object Scalaz {
       compilerPlugin("com.github.tomasmikula" %% "pascal"          % "0.2.1"),
       compilerPlugin("com.github.ghik"        %% "silencer-plugin" % "0.6")
     ),
-    incOptions ~= (_.withLogRecompileOnMacro(false))
+    incOptions ~= (_.withLogRecompileOnMacro(false)),
+    maxErrors := 5
```
Comment (jdegoes) [inline project/ScalazBuild.scala]: Why this one?

Comment (NeQuissimus) [reply project/ScalazBuild.scala]: oops, forgot to take it back out, the output was overwhelming initially :D

File: example/src/main/tut/ct/Functor.md [hunk 3]
```diff
@@ -37,6 +37,6 @@ val listFunctor: Functor[List] = instanceOf(new FunctorClass[List] {
 [Maybe](../data/Maybe.html) does not have a `map` method. Instead its `Functor` is used, which is implicitly available from the above imports.
 
 ```tut
-val x: Maybe[Long] = just(50L)
+val x = just(50L)
```
Comment (hrhino) [inline example/src/main/tut/ct/Functor.md]: Why?

Comment (NeQuissimus) [reply example/src/main/tut/ct/Functor.md]: #1835

**Pull request merged**